backend/geonature/tests/test_validation.py

```diff
@@ -99,3 +129,48 @@ def test_get_validation_history(self, users, synthese_data):
         assert response.status_code == 200
         assert len(response.data) > 0
         assert response.json[0]["id_status"] == str(id_nomenclature_valid_status.id_nomenclature)
+
+    def test_auto_validation(
+        self,
+        users,
+        app,
+        auto_validation_enabled,
+        validation_with_max_score_and_wait_validation_status,
+    ):
+        fct_auto_validation_name = app.config["VALIDATION"][
+            "AUTO_VALIDATION_SQL_FUNCTION"
+        ]  # config["VALIDATION"]["AUTO_VALIDATION_SQL_FUNCTION"]
+        set_logged_user(self.client, users["user"])
+
+        id_nomenclature_probable = db.session.execute(
+            """ select tn.id_nomenclature from ref_nomenclatures.t_nomenclatures tn where tn.mnemonique = 'Probable' """
+        ).scalar()
+        id_nomenclature_attente_validation = db.session.execute(
+            """ select tn.id_nomenclature from ref_nomenclatures.t_nomenclatures tn where tn.mnemonique = 'En attente de validation' """
+        ).scalar()
```

Comment:
**Suggestion:** Consider using a setup function to reduce code repetition and improve readability. [maintainability]
```suggestion
        def get_id_nomenclature(mnemonique):
            return db.session.execute(
                """ select tn.id_nomenclature from ref_nomenclatures.t_nomenclatures tn where tn.mnemonique = :mnemonique """
            ).scalar()

        id_nomenclature_probable = get_id_nomenclature('Probable')
        id_nomenclature_attente_validation = get_id_nomenclature('En attente de validation')
```